### PR TITLE
2357 Refactor RecordTable to use the new scope architecture

### LINKED ISSUE #2357: Refactor RecordTable to use the new scope architecture
## Scope & Context

We are doing a global refactoring to introduce components scopes.

## Technical inputs

- Replace TableContext.Provider with a <RecordTableScope>
```
<TableContext.Provider
          value={{
            onColumnsChange: useRecoilCallback(() => (columns) => {
              setCurrentViewFields?.(columnDefinitionsToViewFields(columns));
              persistViewFields(columnDefinitionsToViewFields(columns));
            }),
          }
>
```
- Create a custom hook useRecordTable inspired by `useView`, `useViewFields`, `useViewFilters`, `useViews` and `useViewSorts`.
- Replace `const { onColumnsChange } = useContext(TableContext);` by `const { onColumnsChange } = useRecordTable()`
- Replace the occurrences of `useRecoilScopedValue` by `useRecoilScopedValueV2`

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #2357

### CODE REVIEW

File: front/src/modules/companies/hooks/useDeleteCompanies.ts [hunk 1]
```diff
@@ -3,15 +3,15 @@ import { useRecoilState, useRecoilValue } from 'recoil';
 
 import { useOptimisticEvict } from '@/apollo/optimistic-effect/hooks/useOptimisticEvict';
```
Comment (charlesBochet) [inline front/src/modules/companies/hooks/useDeleteCompanies.ts:4]: it looks to me that this code is not used anymore?

File: front/src/modules/metadata/components/RecordTableEffect.tsx [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
+import { useRecordTableScopedStates } from '@/ui/object/record-table/hooks/internal/useRecordTableScopedStates';
 import { useView } from '@/views/hooks/useView';
 import { ViewType } from '@/views/types/ViewType';
 
+import { useRecordTable } from '../../ui/object/record-table/hooks/useRecordTable';
 import { useFindManyObjects } from '../hooks/useFindManyObjects';
-import { useObjectMetadataItemInContext } from '../hooks/useObjectMetadataItemInContext';
+import { useFindOneObjectMetadataItem } from '../hooks/useFindOneObjectMetadataItem';
 
 export const RecordTableEffect = () => {
```
Comment (charlesBochet) [inline front/src/modules/metadata/components/RecordTableEffect.tsx:14]: this Component should be renamed to something like RecordTablePageEffect.tsx as we already have RecordTableEffect within the table

File: front/src/modules/metadata/hooks/useSetDataTableData.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
 import { entityFieldsFamilyState } from '@/ui/object/field/states/entityFieldsFamilyState';
-import { useResetTableRowSelection } from '@/ui/object/record-table/hooks/useResetTableRowSelection';
+import { useRecordTable } from '@/ui/object/record-table/hooks/useRecordTable';
 import { isFetchingRecordTableDataState } from '@/ui/object/record-table/states/isFetchingRecordTableDataState';
 import { numberOfTableRowsState } from '@/ui/object/record-table/states/numberOfTableRowsState';
 import { tableRowIdsState } from '@/ui/object/record-table/states/tableRowIdsState';
 import { useView } from '@/views/hooks/useView';
 
 export const useSetObjectRecordTableData = () => {
```
Comment (charlesBochet) [inline front/src/modules/metadata/hooks/useSetDataTableData.ts:10]: the file should also be renamed. and why OBJECTRecordTableData and not RecordTableData?

File: front/src/modules/views/hooks/internal/useViewFilters.ts [hunk 4]
```diff
@@ -21,7 +21,9 @@ export const useViewFilters = (viewScopeId: string) => {
   });
   const apolloClient = useApolloClient();
 
-  const { currentViewFiltersState } = useViewScopedStates();
+  const { currentViewFiltersState } = useViewScopedStates({
+    customViewScopeId: viewScopeId,
```
Comment (charlesBochet) [inline front/src/modules/views/hooks/internal/useViewFilters.ts:25]: why do we need to do this change?

**Review by charlesBochet — APPROVED** [review]
Great! Left comments but minors!

File: front/src/modules/views/hooks/internal/useViewFilters.ts (hunk 4 above)
Comment (bosiraphael) [reply front/src/modules/views/hooks/internal/useViewFilters.ts:25]: Because otherwise useView can't be called from outside of the viewScope I think.

**Pull request merged**